### DISCUSSION TITLE
[Ressource] Adversarial Examples for Deep Neural Networks

### DISCUSSION THREAD

**Discussion by 7ben18** [AdversarialAttacks/.github#9] [Ressourcen]
### Art der Ressource?

- [ ] Paper
- [X] Video
- [ ] etc...

### Quelle?

https://www.youtube.com/watch?v=kxyacmVSGlI&t=294s&ab_channel=PaulHand

### Einleitung
CNN Modelle entscheiden anders als Menschen basierend auf deren Features die sie erhalten. 

Im Video geht es um CNN die Bilder Klassifizieren. Dabei kann man die Mathematische Formuel wie folgt beschreiben: 
![image](https://github.com/AdversarialAttacks/.github/assets/66916399/596e5d4a-e20a-4b5a-b9c1-31935a5d161b)

Parameters sind Weights und Bias die vortrainiert wurden. 
Y gibt uns eine Wahrscheinlichkeitsverteilung für die Klasse zurück (Durch Softmax)

### Arten von Adverserial Attacks
1.  Für das gegebene X suchen wir die (Störung = Petrubations), das Delta δ, welches dann durch den Input X + Delta δ  nicht die entsprechende Ground Truth Value von X ergibt. Dies bezeichnet man als ungezielte Attacke. 
2. Oder wir suchen eine bestimmte Klasse T, und versuchen eine Störung δ zu erhalten, indem wir X+ δ die bestimmte Klasse T ergibt. Dies bezeichnet man als gezielte Attacke. 
![image](https://github.com/AdversarialAttacks/.github/assets/66916399/bbfa125d-3746-438a-859b-9a55df9cd938)

#### Targeted vs. Untargeted
- Entweder wird ein bestimmtes Label (targeted) verändert oder die Attacke führt durch Veränderung, sprich wegen δ zu einer Missklassifikation. 

#### White Box vs. Black Box vs. No Box Attacks
- White Box ist, wenn wir Zugriff auf alle Weights und Bias haben, wir kennen die Architektur des Modelles und können die Backpropagation durchführen.
- Black Box ist, wenn wir keinen Zugriff auf Weights und Bias haben und nicht Backpropagieren können. Hier erhalten wir nur die Wahrscheinlichkeitsverteilung (Vorhhersage des Modelles)
- No Box ist, wenn wir gar keine Infos über das Modelle haben und auch überhaupt keinen Zugriff darauf. 
- Beispiel für Black Box Attacke: Eine Firma bietet den Service an Bilder zu klassifizieren Bspw. Vogelarten oder Pflanzenarten. Hier können wir kein Backpropagation durchführen und somit ist unsere Attacke nur geringfügig möglich. 

#### Impercetible vs. Percetible
- Hier bezieht es sich auf die Wahrnehmungen des Menschen vom Input X für das Modell, wenn δ hinzugefügt wird. 

#### Digital vs. Physical Attacks
- Digitale Attacken ist, wenn der Input Wert am Computer verändert wird
- Physical Attack ist, wenn der Input in der realen Welt verändert wird

#### Specific vs. Universal Attacks
- Specific ist, wenn es nur auf ein einziges spezifischer Klasse funktioniert oder CNN Classifier Modell. 
- Universal ist, wenn es auf alle Input werde funktioniert oder auf alle CNN Classifier

#### Attack vs. Defense
- Attacke ist, wenn man Verursacht, das Modell absichtlich zu einer Fehlklassifikation trifft. 
- Defense ist, wenn man das Modell beim trainieren Robuster gemacht wird, um gegen Attacken robuster zu werden. 

### Wichtigste Keypoints?

- Targeted vs. Untargeted Attacks
- White Box vs. Black Box vs. No Box Attacks

### Motivation von Adverserial Attacks
Falls wir Lebenswichtige Systeme bauen die Machine Learning Algorithmen beinhalten müssen wir uns sicher sein, dass diese nicht einfach durch kleine Veränderungen getäuscht werden können. Modelle sind somit nicht Robust!

### Vokab 
perturbations = Störungen
untargeted = ungezielt
percetible  = warhnehmbar
